### PR TITLE
Pad the examples for QLoRa finetuning test

### PR DESCRIPTION
**Pull request by ckvermaAI**
1. Pad the examples up to max_seq_len of 1024
2. Increase the max_steps (from 5 to 50) and eval_steps (from 3 to 10). Set throughput-related arguments (adjust_throughput, throughput_warmup_steps)
4. Update the fine-tuning test name
5. Also, adjust the reference "eval loss" value for fine-tuning test and "output" for inference test.

Additional updates
6. Enable the eager mode for the test (disable the torch.compile mode for now).
7. Add new requirement for installing the bitsandbytes (from https://github.com/bitsandbytes-foundation/bitsandbytes/tree/multi-backend-refactor)

### CODE REVIEW

**Review by vivekgoe — APPROVED** [review]
LGTM.  @regisss can we add these tests to slow_tests you run for G2/G3? This will help us avoid regressions.  We can publish QLoRA support as experimental/beta future on G2/G3 (works for limited configurations and performance is not great).

**Comment by ckvermaAI** [comment]
Support for NF4 quantization/dequantization using Intel Gaudi hardware: https://github.com/bitsandbytes-foundation/bitsandbytes/pull/1592

File: setup.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
     "torchsde",
     "timm",
     "peft",
+    "bitsandbytes @ git+https://github.com/bitsandbytes-foundation/bitsandbytes.git@multi-backend-refactor",
```
Comment (regisss) [inline setup.py:55]: This is not part of a bitsandbytes release yet?

Comment (ckvermaAI) [reply setup.py:55]: Currently, the support for HPU (Intel Gaudi) and other devices (Intel CPU, Intel GPU, AMD GPUs) is available on multi-backend-refactor branch (https://github.com/bitsandbytes-foundation/bitsandbytes/tree/multi-backend-refactor). That's why we need to install bitsandbytes from specified link.

**Comment by uartie** [comment]
...

```
pip install -r examples/stable-diffusion/requirements.txt
pip install -r examples/stable-diffusion/training/requirements.txt
```
... appears to cause the error to show up during test run.

**Comment by uartie** [comment]
Possibly related to old peft version in `examples/stable-diffusion/training/requirements.txt`... please fix the requirements file.

**Comment by ckvermaAI** [comment]
This error is coming from bitsandbytes, it should not be related to peft. I've run other tests locally (test_bnb_qlora.py, test_bnb_inference.py), and I didn't face this issue. 

Let me check it again.

**Comment by ckvermaAI** [comment]
In case of HPU, bitsandbytes loads the CPU binaries (https://github.com/bitsandbytes-foundation/bitsandbytes/blob/multi-backend-refactor/bitsandbytes/cextension.py#L73), which are not required. 
And if there is no issue in test/topology, you'll not see OSError due to missing *.so file, but if failure occurs due to any other reason, you'll also see the OSError due to missing *.so file.

In short, 
1) I'll move the bitsandbytes installation from setup.py to bitsandbytes tests (test_bnb_qlora.py, test_bnb_inference.py).
2) Later, we'll try to upstream the fix to the bitsandbytes repo.

**Comment by ckvermaAI** [comment]
Fix for the above issue: 
Move bitsandbytes requirements from setup.py to bnb tests
https://github.com/huggingface/optimum-habana/pull/1946
